### PR TITLE
Feat/dark mode exp

### PR DESCRIPTION
**Pull request by FredrikOseberg**
Exploratory PR that adds dark mode to the unleash project. 

Discussion points:
- Color scheme
- Organization of colors and theme structure

#1211

### CODE REVIEW

File: src/contexts/UIContext.ts [hunk 1]
```diff
@@ -27,11 +31,17 @@ const setShowFeedbackPlaceholder = () => {
...
+const setModePlaceHolder = () => {
+    throw new Error('setMode called outside UIContext');
+};
+
 const UIContext = React.createContext<IUIContext>({
     toastData: createEmptyToast(),
     setToast: setToastPlaceholder,
     showFeedback: false,
     setShowFeedback: setShowFeedbackPlaceholder,
+    mode: 'light',
```
Comment (olav) [inline src/contexts/UIContext.ts:43]: Maybe we should use `themeMode: 'light'`? Or just `theme`.

File: src/component/menu/Header/Header.tsx [hunk 2]
```diff
@@ -103,6 +114,17 @@ const Header: VFC = () => {
         );
     }
 
+    const onSetMode = () => {
```
Comment (olav) [inline src/component/menu/Header/Header.tsx:117]: Could move this into `UIContext.setMode` or a custom hook.

File: src/component/menu/Header/Header.tsx [hunk 3]
```diff
@@ -114,9 +136,20 @@ const Header: VFC = () => {
                     )}
                     aria-label="Home"
                 >
-                    <UnleashLogo
-                        className={styles.logo}
-                        aria-label="Unleash logo"
+                    <ConditionallyRender
```
Comment (olav) [inline src/component/menu/Header/Header.tsx:139]: Maybe we should have a helper component for this case:

```tsx
<ThemeMode 
    light={<UnleashLogo ... />} 
    dark={<UnleashLogo ... />} 
/>
```

File: src/component/menu/Header/Header.tsx [hunk 4]
```diff
@@ -151,6 +184,20 @@ const Header: VFC = () => {
                         />
                     </div>
                     <div className={styles.userContainer}>
+                        <ConditionallyRender
```
Comment (olav) [inline src/component/menu/Header/Header.tsx:187]: Would be nice to have three choices (light/dark/system). Not sure if necessary for the first version though.

File: src/component/playground/Playground/Playground.tsx [hunk 5]
```diff
@@ -23,6 +23,7 @@ import Loader from '../../common/Loader/Loader';
 export const Playground: VFC<{}> = () => {
     const { environments } = useEnvironments();
     const theme = useTheme();
+    const [initialLoad, setInitialLoad] = useState(true);
```
Comment (olav) [inline src/component/playground/Playground/Playground.tsx:26]: Looks removable. :thinking:

File: src/component/providers/UIProvider/UIProvider.tsx [hunk 6]
```diff
@@ -1,19 +1,47 @@
...
-const UIProvider: React.FC = ({ children }) => {
+const resolveMode = (darkmode: boolean): themeMode => {
+    const value = getLocalStorageItem('unleash-theme');
+    if (value) {
+        return value as themeMode;
+    }
+
+    let osDark;
+    if (darkmode) {
+        osDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
```
Comment (olav) [inline src/component/providers/UIProvider/UIProvider.tsx:14]: Would be cool if this was a custom `useEffect` hook that automatically updates on OS dark/light mode change.

* https://usehooks.com/useDarkMode/
* https://usehooks.com/useMedia/

File: src/component/providers/UIProvider/UIProviderContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+export const UIProviderContainer: React.FC = ({ children }) => {
+    const { uiConfig } = useUiConfig();
+
+    if (!uiConfig.flags) {
+        return null;
+    }
+
+    return (
+        <UIProvider darkmode={uiConfig.flags.DARK_MODE || false}>
```
Comment (olav) [inline src/component/providers/UIProvider/UIProviderContainer.tsx:12]: Should this be `enableDarkModeSupport` or something like that? Or does it actually toggle dark mode as well?

File: src/component/menu/Header/Header.tsx (hunk 3 above)
Comment (FredrikOseberg) [reply src/component/menu/Header/Header.tsx:139]: Great idea

File: src/component/menu/Header/Header.tsx (hunk 2 above)
Comment (FredrikOseberg) [reply src/component/menu/Header/Header.tsx:117]: Might make sense when/if we end up using it in multiple places

**Review by olav — APPROVED** [review]

**Pull request merged**